src/huggingface_hub/hf_api.py

```diff
@@ -1139,15 +1139,22 @@ def save_token(cls, token):
             f.write(token)
 
     @classmethod
-    def get_token(cls):
+    def get_token(cls) -> Optional[str]:
         """
-        Get token or None if not existent.
+        Get token or None if not existent. A token can be also provided using env variables
+
+        >>> export HUGGING_FACE_HUB_TOKEN=<YOUR_TOKEN>
         """
-        try:
-            with open(cls.path_token, "r") as f:
-                return f.read()
-        except FileNotFoundError:
-            pass
+
+        path_token: Path = Path(cls.path_token)
+        token: Optional[str] = os.environ.get("HUGGING_FACE_HUB_TOKEN")
+
+        if token is None:
+            # fall back to disk
+            if path_token.exists():
+                with path_token.open("r") as f:
+                    token = f.read()
```

Comment:
Actually, the try/except block was more pythonic here. I don't think it's a huge deal, but the try/except is more reliable and more up-to-date than the check with the `if path_token.exists()` which can return information that will be different by the time it is consumed.

The approach you implemented is based off of the [look-before-you-leap](https://docs.python.org/2.7/glossary.html#term-lbyl) paradigm, but in Python we favor the [easier-to-ask-for-forgiveness-than-permission](https://docs.python.org/2.7/glossary.html#term-eafp) paradigm.

Would you mind reverting this change? Other than that, feel free to merge :)

Reply:
Reverted :)